core/src/main/java/fr/sncf/osrd/standalone_sim/SafetySpeed.kt

```diff
@@ -55,6 +74,20 @@ fun makeSafetySpeedRanges(
     return res.subMap(0.meters, chunkPath.length)
 }
 
+/**
+ * Create a safety speed range at the end of the last route, either short slip or normal stop
+ * depending on whether it ends at a buffer stop.
+ */
+private fun makeEndOfPathStop(
+    infra: RawSignalingInfra,
+    routes: StaticIdxList<Route>,
+    signalOffsets: List<Offset<TravelledPath>>
+): SafetySpeedStop {
+    val lastRouteExit = infra.getRouteExit(routes.last())
+    val isBufferStop = infra.isBufferStop(lastRouteExit.value)
+    return SafetySpeedStop(signalOffsets.last(), isBufferStop)
+}
```

Comment:
Question: do we always want to add safety speeds at the end of the last route, or just if it ends with a buffer stop?

I don't think this was explicitly mentioned anywhere, but I believe it makes sense. We only allocate ressources up to the end of the last route, which means that we always assume it ends with a blocking signal. 

I can change it if we only want that to happen on buffer stop. But I believe there would be an inconsistency somewhere if we do that.

Reply:
Actually, that doesn't work: sometimes we don't want the train to stop at the end of its simulation. The last stop should generally already be "on stop signal" anyway.

I'll change that.